arch/sim/src/sim/up_oneshot.c

```diff
@@ -376,25 +437,13 @@ void up_timer_initialize(void)
 
 void up_timer_update(void)
 {
-  static const struct timespec tick =
-  {
-    .tv_sec  = 0,
-    .tv_nsec = NSEC_PER_TICK,
-  };
-
-  FAR sq_entry_t *entry;
-
-  clock_timespec_add(&g_current, &tick, &g_current);
+#ifdef CONFIG_SIM_WALLTIME_HOSTSLEEP
 
-#ifdef CONFIG_SIM_WALLTIME
   /* Wait a bit so that the timing is close to the correct rate. */
 
   host_sleepuntil(g_current.tv_nsec +
     (uint64_t)g_current.tv_sec * NSEC_PER_SEC);
-#endif
 
-  for (entry = sq_peek(&g_oneshot_list); entry; entry = sq_next(entry))
-    {
-      sim_process_tick(entry);
-    }
+  sim_timer_update();
```

Comment:
Need call sim_timer_update when CONFIG_SIM_WALLTIME=n
```
#ifdef CONFIG_SIM_WALLTIME_HOSTSLEEP
host_sleepuntil(g_current.tv_nsec +
    (uint64_t)g_current.tv_sec * NSEC_PER_SEC);
#endif
#ifndef CONFIG_SIM_WALLTIME_SIGNAL
sim_timer_update();
#endif
```

Reply:
Yes, I saw that (sim_timer_update was not called). Thanks